File: fm-domain/src/main/java/io/zeitwert/fm/collaboration/model/ObjNote.java
package io.zeitwert.fm.collaboration.model;

import io.dddrive.core.ddd.model.Aggregate;
import io.dddrive.core.obj.model.Obj;
import io.zeitwert.fm.collaboration.model.enums.CodeNoteType;

public interface ObjNote extends Obj {

	Object getRelatedToId();

	void setRelatedToId(Object id);

	Aggregate getRelated();

	CodeNoteType getNoteType();

	void setNoteType(CodeNoteType noteType);

	String getSubject();

	void setSubject(String subject);

	String getContent();

	void setContent(String subject);

	Boolean getIsPrivate();

	void setIsPrivate(Boolean isPrivate);

}

File: fm-domain/src/main/java/io/zeitwert/fm/collaboration/model/ObjNote.kt
package io.zeitwert.fm.collaboration.model

import dddrive.app.ddd.model.Aggregate
import dddrive.app.obj.model.Obj
import io.zeitwert.fm.collaboration.model.enums.CodeNoteType

interface ObjNote : Obj {

	var relatedToId: Any?

	val relatedTo: Aggregate?

	var noteType: CodeNoteType?

	var subject: String?

	var content: String?

	var isPrivate: Boolean?

}
